src/autogluon/bench/eval/scripts/run_generate_clean_openml.py

```diff
@@ -1,69 +1,75 @@
+import typer
 
 from autogluon.bench.eval.evaluation.constants import FRAMEWORK
 from autogluon.bench.eval.evaluation.preprocess import preprocess_openml
 from autogluon.common.savers import save_pd
 
+app = typer.Typer()
+logger = logging.getLogger(__name__)
 
-def clean_and_save_results(
-    run_name,
-    results_dir="data/results/",
-    results_dir_input=None,
-    results_dir_output=None,
-    file_prefix="results_automlbenchmark",
-    run_name_in_input_path=True,
-    constraints=None,
-    out_path_prefix="openml_ag_",
-    out_path_suffix="",
-    framework_suffix_column="constraint",
+
+@app.command()
+def clean_amlb_results(
+    benchmark_name: str = typer.Option(
+        ..., help="Benchmark name populated by benchmark run, in format <benchmark_name>_<timestamp>"
+    ),
+    results_dir: str = typer.Option("data/results/", help="Root directory of raw and prepared results."),
+    results_input_dir: str = typer.Option(
+        None,
+        help="Directory of the results file '<file_prefix><constraint_str><benchmark_name_str>.csv' getting cleaned. Can be an S3 URI. If not provided, it defaults to '<results_dir>input/raw/'",
+    ),
+    results_output_dir: str = typer.Option(
+        None,
+        help="Output directory of cleaned file. Can be an S3 URI. If not provided, it defaults to '<results_dir>input/prepared/openml/'",
+    ),
+    file_prefix: str = typer.Option("results_automlbenchmark", help="File prefix of the input results files."),
+    benchmark_name_in_input_path: bool = False,
+    constraints: str = typer.Option(
+        None,
+        help="List of AMLB constraints, refer to https://github.com/openml/automlbenchmark/blob/master/resources/constraints.yaml",
+    ),
+    out_path_prefix: str = typer.Option("openml_ag_", help="Prefix of result file."),
+    out_path_suffix: str = typer.Option("", help="suffix of result file."),
 ):
-    if results_dir_input is None:
-        results_dir_input = results_dir + "input/raw/"
-    if results_dir_output is None:
-        results_dir_output = results_dir + "input/prepared/openml/"
-    run_name_str = f"_{run_name}" if run_name_in_input_path else ""
+    """
+    Cleans and aggregate results further with unified column names and adds benchmark name into framework column.
 
-    results_list = []
+    Example:
+        agbench clean-and-save-results --benchmark-name ag_tabular_20230629T140546 --results-input-dir s3://autogluon-benchmark-metrics/aggregated/tabular/ag_tabular_20230629T140546/ --benchmark-name-in-input-path
+    """
+
+    if results_input_dir is None:
+        results_input_dir = os.path.join(results_dir, "input/raw/")
+    if results_output_dir is None:
+        results_output_dir = os.path.join(results_dir, "input/prepared/openml/")
+
+    benchmark_name_str = f"_{benchmark_name}" if benchmark_name_in_input_path else ""
     if constraints is None:
         constraints = [None]
+    else:
+        constraints = constraints.split(",")
```

Comment:
This is problematic. So I can't call this function with pure python anymore and pass a list?

Reply:
I kept the original function, and have added a `typer` wrapper to take `constraints` as list by providing `--constraints constraint_1 --constraints constraint_2`.